pkg/api/handlers/compat/secrets.go

```diff
@@ -0,0 +1,107 @@
+package compat
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"net/http"
+
+	"github.com/containers/podman/v2/libpod"
+	"github.com/containers/podman/v2/pkg/api/handlers/utils"
+	"github.com/containers/podman/v2/pkg/domain/entities"
+	"github.com/containers/podman/v2/pkg/domain/infra/abi"
+	"github.com/gorilla/schema"
+	"github.com/pkg/errors"
+)
+
+func ListSecrets(w http.ResponseWriter, r *http.Request) {
+	var (
+		runtime = r.Context().Value("runtime").(*libpod.Runtime)
+	)
+
+	ic := abi.ContainerEngine{Libpod: runtime}
+	reports, err := ic.SecretList(r.Context())
+	if err != nil {
+		utils.InternalServerError(w, err)
+		return
+	}
+	utils.WriteResponse(w, http.StatusOK, reports)
+}
+
+func InspectSecret(w http.ResponseWriter, r *http.Request) {
+	var (
+		runtime = r.Context().Value("runtime").(*libpod.Runtime)
+	)
+	name := utils.GetName(r)
+	names := []string{name}
+	ic := abi.ContainerEngine{Libpod: runtime}
+	reports, errs, err := ic.SecretInspect(r.Context(), names)
+	if err != nil {
+		utils.InternalServerError(w, err)
+		return
+	}
+	if errs != nil {
+		utils.SecretNotFound(w, name, errs[0])
+		return
+	}
+	utils.WriteResponse(w, http.StatusOK, reports[0])
+}
+
+func RemoveSecret(w http.ResponseWriter, r *http.Request) {
+	var (
+		runtime = r.Context().Value("runtime").(*libpod.Runtime)
+	)
+
+	opts := entities.SecretRmOptions{}
+	name := utils.GetName(r)
+	ic := abi.ContainerEngine{Libpod: runtime}
+	reports, err := ic.SecretRm(r.Context(), []string{name}, opts)
+	if err != nil {
+		utils.InternalServerError(w, err)
+		return
+	}
+	if reports[0].Err != nil {
+		utils.SecretNotFound(w, name, reports[0].Err)
+		return
+	}
+	utils.WriteResponse(w, http.StatusNoContent, nil)
+}
+
+func CreateSecret(w http.ResponseWriter, r *http.Request) {
```

Comment:
This doesn't look like it's compatible with the Docker API - https://docs.docker.com/engine/api/v1.40/#operation/SecretCreate

Docker supports more parameters, uses no query parameters, and sends everything JSON-encoded in the request body.

If we're not actually Docker-compat, we should not be exposing the compat endpoints.

Reply:
I think I just accidentally wrote extra code for the query params - the handler doesn't actually use it. The only thing missing is labels, which we decided not to implement as of now. I don't think docker actually accepts the template field, it looks to be a docs error, a copy of the driver field. I could either remove the docker-compat or do a logrus warning if someone tries to use labels, let me know the right way to go

Reply:
We usually return an error when they pass a non-supported field

Reply:
Alright, so then is removing the docker-compat the way to go, and making it a libpod-only endpoint? It would also make sense since docker secrets only exist in the swarm and podman secrets are different enough since they work in vanilla containers.

Reply:
If the endpoint accepts the same parameters, we can keep the `compat` endpoint, as long as it returns a Bad Request when something we do not support is requested